src/operator/tensor/broadcast_reduce_op.h

```diff
@@ -556,6 +556,162 @@ inline bool ReduceAxesOpForwardStorage(const nnvm::NodeAttrs& attrs,
   return dispatched;
 }
 
+struct argmax {
+  template<typename DType>
+  MSHADOW_XINLINE static void Map(int i, const int nWorkers, const DType *in_data, DType *out_data,
```

Comment:
Please use `index_t` for Map index so that we can support large tensor.

Reply:
OK, I will do that.

Reply:
OK, I will do that.